bin/metrics-rabbitmq-queue.rb

```diff
@@ -120,16 +138,21 @@ def run
       queue['messages'] ||= 0
       drain_time = queue['messages'] / queue['backing_queue_status']['avg_egress_rate']
       drain_time = 0 if drain_time.nan? || drain_time.infinite? # 0 rate with 0 messages is 0 time to drain
-      output([config[:scheme], queue['name'], 'drain_time'].join('.'), drain_time.to_i, timestamp)
-
-      %w(messages consumers).each do |metric|
-        output([config[:scheme], queue['name'], metric].join('.'), queue[metric], timestamp)
+      queue['drain_time'] = drain_time.to_i
+
+      metrics = dotted_keys(queue)
+      metrics.each do |metric|
+        next unless metric.match(config[:metrics])
+        value = queue.dig(*metric.split('.'))
+        # Special case of *gress rates for backward-compatibility
+        if metric =~ 'backing_queue_status.avg'
+          value = format('%.4f', value)
+          metric = metric.split('.')[-1]
+        end
+        output("#{config[:scheme]}.#{queue['name']}.#{metric}", value, timestamp) unless value.nil?
```

Comment:
@majormoses I realize while re-reading this that I could have kept the `array.join('.')` pattern, there is no reason for using placeholders other than that's what I am used to. Let me know if you want to keep the existing style.